docs/deployment-integration.md

```diff
@@ -0,0 +1,312 @@
+  --admin $ADMIN_ADDRESS \
+  --fees 100000ubbn \
+  -y
+```
+
+The parameters:
+* `admin`: The Babylon Genesis address of the contract administrator
+* `bsn_id`: The unique identifier for the BSN rollup
+  Ensure that this is unique and not already used by another BSN on Babylon
+  Genesis as this will affect your registration
+* `min_pub_rand`: Minimum public randomness values included per public
+  randomness commit by finality providers
+* `rate_limiting_interval`: Length (in blocks) of the interval for
+  rate-limiting submissions
+* `max_msgs_per_interval`: Maximum number of submissions allowed per finality
+  provider within each interval
+* `allowed_finality_providers`: List of finality providers from which the contract will
+  accept finality signatures and randomness
+
+<!-- TODO: once we have docs for the allow list and rate limiting, we can link
+those here -->
+
+## Rollup BSN Registration
+<img width="3032" height="287" alt="register" src="https://github.com/user-attachments/assets/935fb9bd-f42d-47ba-a0f3-0fc628961e67" />
+
+After deploying and instantiating the Rollup BSN contract, the
+rollup must register on Babylon Genesis. Registration requires 
+submitting metadata to identify and describe the BSN
+
+To register the BSN, use:
+```shell
+babylond tx btcstkconsumer register-consumer \
+                                <consumer_id> \
+                                <consumer_name> \
+                                <consumer_ description> \
+                                <max_multi_staked_fps> \
+                                <rollup_finality_contract_address>
+```
+
+Required metadata for BSN registration:
+* `consumer_id`: BSN ID of your rollup (same as the one you used on contract
+  instantiation)
+* `consumer_name`: Human-readable name of your rollup BSN (e.g., `"DeFi Rollup Chain"`)
+* `consumer_description`: Brief description of the rollup BSN's purpose
+* `max_multi_staked_fps`: Maximum number of rollup FPs per BTC delegation. Prevents delegators from splitting
+ their BTC across too many networks (recommended: `3-5`)
+<!-- TODO: this should not be here, we will have a global limit on fps -->
+* `rollup_finality_contract_address`: Babylon Genesis address of the deployed Rollup BSN contract (`bbn1...` format)
+
+> **Note**: The above operation requires governance in permissioned
+> registration networks.
+
+## Contract Maintenance
+<img width="1848" height="677" alt="adminRole" src="https://github.com/user-attachments/assets/9b942068-3d72-4ae8-a47e-79894427204e" />
+
+The Rollup BSN contract exposes `admin-only` functions for ongoing management. These allow you to clean up
+ old data, update roles, and adjust configuration without redeploying the contract.
+
+Each operation is executed with:
+
+```shell
+babylond tx wasm execute <CONTRACT_ADDRESS> '<MSG>' \
+  --from $ADMIN_KEY \
+  --chain-id $BBN_CHAIN_ID \
+  --keyring-backend test \
+  --gas auto \
+  --gas-adjustment 1.3 \
+  --fees 1000000ubbn \
+  -y
```

Comment:
These parameteres need to be cleaned up, specifically:
* --gas auto + --gas-adjustment + --fees do not make sense to go together.
* `-y` avoids a safety check
* `--keyring-backend test` assumes that the keyring is a test keyring.

I suggest we either don't provide those parameters at all (i.e., we just list `babylond tx wasm execute <CONTRACT_ADDRESS> '<MSG>'`) or we explain them. Given that this is not a guide on how to send cosmos txs, I suggest we do the former and keep things simple.

Same for all other commands.